node/src/components/deploy_buffer.rs

```diff
@@ -76,26 +80,74 @@ impl Display for Event {
     }
 }
 
+type DeployCollection = HashMap<DeployHash, DeployHeader>;
+type ProtoBlockCollection = HashMap<ProtoBlockHash, DeployCollection>;
+/// Deploy buffer.
+#[derive(DataSize, Debug, Default, Clone)]
+struct DeployBufferInner {
+    block_max_deploy_count: usize,
+    collected_deploys: DeployCollection,
+    processed: ProtoBlockCollection,
+    finalized: ProtoBlockCollection,
+}
+
 /// Deploy buffer.
 #[derive(DataSize, Debug, Clone)]
 pub(crate) struct DeployBuffer {
-    block_max_deploy_count: usize,
-    collected_deploys: HashMap<DeployHash, DeployHeader>,
-    processed: HashMap<ProtoBlockHash, HashMap<DeployHash, DeployHeader>>,
-    finalized: HashMap<ProtoBlockHash, HashMap<DeployHash, DeployHeader>>,
+    inner: Arc<Mutex<DeployBufferInner>>, // Don't hand out references to internal mutable state
 }
 
 impl DeployBuffer {
     /// Creates a new, empty deploy buffer instance.
     pub(crate) fn new(block_max_deploy_count: usize) -> Self {
-        DeployBuffer {
+        let inner = Arc::new(Mutex::new(DeployBufferInner {
             block_max_deploy_count,
-            collected_deploys: HashMap::new(),
-            processed: HashMap::new(),
-            finalized: HashMap::new(),
-        }
+            ..Default::default()
+        }));
+        let weak = Arc::downgrade(&inner);
+        tokio::spawn(async move {
+            while let Some(buffer) = weak.upgrade() {
+                let pruned_items = buffer.lock().unwrap().prune();
+                log::info!("{} items pruned from deploy buffer", pruned_items);
```

Comment:
info! seems unnecessarily chatty to me; under normal operation I don't think this is all that useful as output.

Reply:
Got it, I'll make the equivalent log line debug!